### PR TITLE
chore: publishable generator

### PR DESCRIPTION
**Pull request by alexanderczigler**
This PR adds an nx plugin that is used to ensure a package has a proper package.json

### CODE REVIEW

**Review by believer — APPROVED** [review]

File: packages/sparkify/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable */
+export default {
+  displayName: 'sparkify',
+  preset: '../../jest.preset.js',
+  transform: {
+    '^.+\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
+  },
+  moduleFileExtensions: ['ts', 'js', 'html'],
+  coverageDirectory: '../../coverage/packages/sparkify',
```
Comment (believer) [inline packages/sparkify/jest.config.ts:9]: This configuration throws an error. I solved it by [moving it to the global configuration](https://github.com/seb-neo/sebspark/commit/857642d069694f172ad13e2cafdc00b6265f109e), so you can remove this line.

File: packages/sparkify/src/executors/ensure/executor.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+export default async function ensureExecutor(
+  options: EnsureExecutorOptions
+): Promise<{ success: boolean }> {
+  process.stdout.write(
+    `Ensuring @${options.organisation}/${options.package} package...`
+  )
+
+  let success = false
+  let packageJson = {} as any
```
Comment (believer) [inline packages/sparkify/src/executors/ensure/executor.ts:16]: This `any` throws a lint warning. You can solve it by adding a type for `package.json`, something like:

```ts
// Add this outside the executor
export type PackageJson = {
  name?: string
  version?: string
  publishConfig?: {
    directory?: string
  }
}
```

```suggestion
  let packageJson: PackageJson = {}
```

File: packages/sparkify/jest.config.ts (hunk 1 above)
Comment (believer) [inline packages/sparkify/jest.config.ts:9]: This configuration throws an error. I solved it by [moving it to the global configuration](https://github.com/seb-neo/sebspark/commit/857642d069694f172ad13e2cafdc00b6265f109e), so you can remove this line.

```suggestion
```

**Comment by believer** [comment]
Something happens with `test-all`/`lint-all` when there's a `project.json` in the root. It goes into an infinite loop. Not sure how to solve it...

File: tools/scripts/ensure.mjs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const packageJsonPath = `${currentDir}/${options.path}/${options.package}/package.json`
+
+  if (!fs.existsSync(packageJsonPath)) {
+    // Initiate a new package.json
+    packageJson = {
+      name: `@${options.organisation}/${options.package}`,
+      version: '0.0.1',
+    }
+  } else {
+    packageJson = JSON.parse(await fs.readFileSync(packageJsonPath))
```
Comment (believer) [inline tools/scripts/ensure.mjs:25]: No need for `await` here.

File: tools/scripts/ensure.mjs [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import * as fs from 'fs'
+
+console.log(process.argv)
```
Comment (believer) [inline tools/scripts/ensure.mjs:3]: Do we need to print `argv`?

**Comment by believer** [comment]
Nice, I think this solution is much cleaner! There's a lot of boilerplate for `nx` stuff 😅

File: README.md [hunk 5]
```diff
@@ -5,3 +5,15 @@
...
+## Contributing
+
+### Adding a new package
+
+Packages are added in two steps; running the appropriate nx generator and running the `sparkify` plugin to ensure the newly added package has a valid `package.json`.
+
+```shell
+# This example created a package @sebspark/meow
+yarn nx generate @nx/node:library meow --publishable --importPath=@sebspark/meow --no-interactive
+yarn ensure -- meow
```
Comment (believer) [inline README.md:18]: With `yarn` we can remove the `--` here and still get the same result.

File: tools/scripts/ensure.mjs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  package: process.argv[2],
+  path: 'packages',
+}
+
+export default async function ensure(options) {
+  console.log(`Ensuring @${options.organisation}/${options.package} package...`)
+
+  let packageJson = {}
+  const currentDir = process.cwd()
+  const packageJsonPath = `${currentDir}/${options.path}/${options.package}/package.json`
```
Comment (believer) [inline tools/scripts/ensure.mjs:16]: What are your thoughts on adding a check if the package exists here? Something like:

```js
const packagePath = `${currentDir}/${options.path}/${options.package}`
const packageJsonPath = `${packagePath}/package.json`

if (!fs.existsSync(packagePath)) {
  console.error(
    `ERROR: @${options.organisation}/${options.package} package does not exist.`
  )
  return
}
```

That would display a prettier error if something doesn't exist.

File: tools/scripts/ensure.mjs (hunk 4 above)
Comment (alexanderczigler) [reply tools/scripts/ensure.mjs:3]: No! ☺️

File: tools/scripts/ensure.mjs (hunk 6 above)
Comment (alexanderczigler) [reply tools/scripts/ensure.mjs:14]: Yes that makes sense, good idea.

**Pull request merged**